src/DataMapper/DataMapper.php

```diff
@@ -108,7 +110,16 @@ public function mapFormsToData($forms, &$data): void
 
             try {
                 if ($forwardToWrappedDataMapper) {
-                    $this->dataMapper->mapFormsToData([$form], $data);
+                    $customPropertyPath = $form->getConfig()->getOption('property_path');
```

Comment:
I do not really understand why we need to change the `DataMapper` class this way. Can you elaborate a bit on it?

Reply:
The `DataMapper` will not handle the custom property path. The `property` or `array key` name will not be custom one.

Reply:
I am not sure I understand what you mean with that. But that's not a big deal I guess. I can check out the changes locally when the PR is ready and debug a bit what's happening here.

Reply:
Yes, it might be easier to understand. There are tests covering it, so it's easy to isolate

Reply:
I think I found out how we can improve this part and I have sent a PR with the changes that I would implement to your fork: dragosprotung/rich-model-forms-bundle#1